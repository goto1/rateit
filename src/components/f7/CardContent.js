import React from "react";
import PropTypes from "prop-types";

const CardContent = ({ className, children }) => {
  const classNames = `card-content ${className ? className : ""}`.trim();
  return (
    <div className={classNames}>
      <div className="card-content-inner">
        {children}
      </div>
    </div>
  );
};

CardContent.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

export default CardContent;
