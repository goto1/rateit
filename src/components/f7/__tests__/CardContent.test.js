import React from "react";
import CardContent from "../CardContent";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

describe("CardContent component", () => {
  it("should render correctly", () => {
    const tree = renderer
      .create(<CardContent>HelloWorld!</CardContent>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render a span", () => {
    const wrapper = shallow(
      <CardContent>
        <span>HelloWorld!</span>
      </CardContent>
    );
    const actual = wrapper.find("span").at(0).html();
    const expected = "<span>HelloWorld!</span>";
    expect(actual).toBe(expected);
  });

  it("should render with a specified className", () => {
    const wrapper = shallow(
      <CardContent className="custom">HelloWorld!</CardContent>
    );
    const actual = wrapper.props().className;
    const expected = "card-content custom";
    expect(actual).toBe(expected);
  });
});
