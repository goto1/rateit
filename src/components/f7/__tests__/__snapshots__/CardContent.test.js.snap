// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContent component should render correctly 1`] = `


<div
  className="card-content"
>
  <div
    className="card-content-inner"
  >
    HelloWorld!
  </div>
</div>
`;
